fhir-core/src/main/java/com/ibm/fhir/core/context/impl/FHIRPagingContextImpl.java

```diff
@@ -24,12 +26,16 @@
      * <pre>
      * page number: 1
      * page size: 10
+     * max page size: 1000
+     * max page include count: 1000
      * last page: 214748364
      * </pre>
      */
     public FHIRPagingContextImpl() {
         this.pageNumber = FHIRConstants.FHIR_PAGE_NUMBER_DEFAULT;
         this.pageSize = FHIRConstants.FHIR_PAGE_SIZE_DEFAULT;
+        this.maxPageSize = FHIRConstants.FHIR_PAGE_SIZE_DEFAULT_MAX;
```

Comment:
Does maxPageSize need to be part of the context? The business logic constructing the FHIRPagingContextImpl should determine the pageSize. My preference would be to not embed business logic inside a context object (which is really just a data-carrier).

Reply:
I added it there as it was a convenient place, since it was calculated when the FHIRPagingContextImpl is being constructed, and it used in several places, I thought holding onto the value here would be ok.  Otherwise, I can change it to read from the config parameter in the handful of places that need it; just thought doing it in one place would be cleaner.  Do you want me to change it?

Reply:
When the FHIRPagingContextImpl is contructed is when the maxPageSize and pageSize are set; at which time, the pageSize is set to be no larger than the maxPageSize. Then based on the `_count` parameter, the maxPageSize is checked, so the pageSize is never set to larger than the maxPageSize.